crates/engine/src/prepare/error.rs

```diff
@@ -38,6 +40,8 @@ impl From<PrepareError> for GraphqlError {
             PrepareError::Plan { err, .. } => err.into(),
             PrepareError::Solve { err, .. } => err.into(),
             PrepareError::NormalizationError => GraphqlError::new(err.to_string(), ErrorCode::InternalServerError),
+            // TODO: Is this a good error code? Not sure
```

Comment:
Think should be `OperationValidationError`

Reply:
for errors in the binding, operation limits we use `OperationValidationError`. `BadRequest` is used for stuff like improper content-type, ill-formed graphql request, etc.